jwst/straylight/straylight.py

```diff
@@ -1,290 +1,180 @@
     output = input_model.copy()  # this is used in algorithm to
+    # find the straylight correction.
+
+    # mask is same size as image - set = 1 everywhere to start
+    mask = np.ones_like(output.data)
 
-    # kernel matrix
-    w = shepard_2d_kernel(roi, power)
-    # mask is same size as slice_map - set = 0 everywhere
-    mask = np.zeros_like(slice_map)
-    # mask = 1 for slice gaps
-    mask[slice_map == 0] = 1
-
-    # find if any of the gap pixels are bad pixels - if so mark them
-    # stick with dq init mask file flags
-    mask_dq = input_model.dq.copy()
-    all_flags = (dqflags.pixel['DEAD'] + dqflags.pixel['HOT'] +
-                 dqflags.pixel['OPEN'] + dqflags.pixel['ADJ_OPEN'] +
-                 dqflags.pixel['RC'] + dqflags.pixel['REFERENCE_PIXEL'])
+    # if there are nans remove them because they mess up the correction
+    index_inf = np.isinf(output.data).nonzero()
+    output.data[index_inf] = 0.0
+    index_inf = np.isnan(output.data).nonzero()
+    output.data[index_inf] = 0.0
+    # flag associated mask so we do not  use any
+    # slice gaps that are nans, now data=0.
+    mask[index_inf] = 0
+
+    # flag pixels that should not be used for computation
+    mask_dq = input_model.dq.copy()  # * mask # find DQ flags of the gap values
+    all_flags = (dqflags.pixel['DO_NOT_USE'] + dqflags.pixel['NON_SCIENCE']
+                 + dqflags.pixel['DEAD'] + dqflags.pixel['HOT'])
+    # where are pixels set to any one of the all_flags cases
     testflags = np.bitwise_and(mask_dq, all_flags)
     # where are testflags ne 0 and mask == 1
-    bad_flags = np.where((testflags != 0) & (mask == 1))
+    bad_flags = np.where(testflags != 0)
     mask[bad_flags] = 0
 
-    # find location of refpixels
-    refpixel = np.bitwise_and(mask_dq, dqflags.pixel['REFERENCE_PIXEL'])
-    index_refpixel = np.where(refpixel != 0)
-
-    # apply mask to the data
-    image_gap = output.data * mask
-
-    # avoid cosmic ray contamination. This check is used to screen out
-    # undetected cosmic rays. Only use the science data for this cosmic ray test.
-    cosmic_ray_test = 0.02 * np.max(output.data[slice_map > 0])
-    image_gap[image_gap > cosmic_ray_test] = 0
-
-    image_gap[image_gap < 0] = 0   # set pixels less than zero to 0
-    image_gap = convolve(image_gap, Box2DKernel(3))   # smooth gap pixels
-    image_gap *= mask   # reset science pixels to 0
-    # we do not want the reference pixels to be used in the convolution
-    image_gap[index_refpixel] = 0.0
-
-    # convolve gap pixel image with weight kernel
-    astropy_conv = convolve(image_gap, w)
-
-    # normalize straylight flux by weights
-    norm_conv = convolve(mask, w)
-
-    astropy_conv /= norm_conv
-
+    # Which channel/band are we using?
+    channel = input_model.meta.instrument.channel
+    band = input_model.meta.instrument.band
+
+    # Deal with normal cases only, we won't apply to cross-dichroic cases for now
```

Comment:
Should there be a final else clause to act as a trap to handle cross-dichroic cases? i.e. issue a warning and skip the step

Reply:
I think a final else clause might catch things I didn't intend given how the if statements are set up, but I've added a default setting before the if statements, an informative message if that doesn't get changed by the if statements, and the try/except should catch any problems.